src/validation.cpp

```diff
@@ -1277,14 +1277,27 @@ void static InvalidChainFound(CBlockIndex* pindexNew)
     CheckForkWarningConditions();
 }
 
+/**
+ * Mark a single block as invalid, updating its status in `mapBlockIndex`.
+ * Perform other index-related bookkeeping.
+ *
+ * See also: InvalidateBlock, which should be called instead of this if the
+ * block being invalidated is on the active chain.
+ */
 void CChainState::InvalidBlockFound(CBlockIndex *pindex, const CValidationState &state) {
-    if (!state.CorruptionPossible()) {
-        pindex->nStatus |= BLOCK_FAILED_VALID;
-        g_failed_blocks.insert(pindex);
-        setDirtyBlockIndex.insert(pindex);
-        setBlockIndexCandidates.erase(pindex);
-        InvalidChainFound(pindex);
+    AssertLockHeld(cs_main);
+
+    if (state.CorruptionPossible()) {
+        LogPrintf("%s: can't be called when corruption is possible (invalid block=%s)",
```

Comment:
Why log this unconditionally? I think it may be trivial to cause someone's logfile to be filled with these.

Reply:
Yep, that's a good point. I'll change this into a comment.